src/TestSessionEnvironment.php

```diff
@@ -578,6 +578,10 @@ public function connectToDatabase($state = null)
                     $this->oldDatabaseName = $databaseConfig['database'];
                     $databaseConfig['database'] = $state->database;
                     DB::connect($databaseConfig);
+                    // This is required for CMS 5.4+
+                    if (method_exists(DB::class, 'hasConfig') && DB::hasConfig('primary')) {
+                        DB::connect($databaseConfig, 'primary');
+                    }
```

Comment:
We're connecting twice?
Shouldn't this be an if/else?

Reply:
Updated